src/lib.rs

```diff
@@ -229,6 +231,139 @@ impl<'a> Into<Device> for &'a str {
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
 pub struct Linktype(pub i32);
 
+#[derive(Debug, PartialEq, Eq, Clone, Copy, IntoPrimitive, TryFromPrimitive)]
+#[repr(i32)]
+pub enum LinkName {
```

Comment:
Although the definition of the `enum LinkName` is beautiful, I'm opposed to adding this because
- It represents the same thing as the existing `struct Linktype`. Providing two `pub` data types on the same concepts look confusing to me.
- It cannot represent `Linktype(290)` which might come from external / future pcap files as you have already pointed out.

My suggestion is to use the following associated `const`s instead:
```
impl Linktype {
    const NULL: Self = Linktype(0);
    const ETHERNET: Self = Linktype(1);
    ...
```

This pattern is also seen in [IPv6Addr::LOCALHOST](https://doc.rust-lang.org/std/net/struct.Ipv6Addr.html#associatedconstant.LOCALHOST) and other locations.

Reply:
That not a bad idea

Reply:
That not a bad idea